examples/pytorch/mnist_trainer.py

```diff
@@ -63,54 +64,72 @@ def parse_args():
     return parser.parse_args()
 
 
-def run_trainer(model, data_module, run_type, plugin=[], devices=1):
-    """Run trainer.fit and trainer.test with given parameters."""
-    _devices = devices
-    _strategy = HPUParallelStrategy(start_method="spawn") if _devices > 1 else SingleHPUStrategy()
-    if run_type == "recipe_caching":
-        os.environ["PT_HPU_RECIPE_CACHE_CONFIG"] = "tmp/recipes,True,1024"
+@contextmanager
+def set_env_vars(env_dict):
+    """CM to set env."""
+    active = bool(env_dict)  # Set to True if env_dict is not None or empty
+    original_values = {key: os.environ.get(key, None) for key in env_dict} if active else {}
+
+    with suppress(Exception):
+        if active:
+            os.environ.update(env_dict)
+        yield
+
+    # Restore env
+    for key, original_value in original_values.items():
+        if original_value is not None:
+            os.environ[key] = original_value
+        else:
+            del os.environ[key]
+
+
+def run_trainer(model, data_module, plugin, devices=1, strategy=None):
+    """Run trainer.fit with given parameters."""
     trainer = Trainer(
         accelerator=HPUAccelerator(),
-        devices=_devices,
-        strategy=_strategy,
+        devices=devices,
+        strategy=strategy if strategy is not None else HPUParallelStrategy() if devices > 1 else SingleHPUStrategy(),
```

Comment:
```suggestion
        strategy=strategy or HPUParallelStrategy() if devices > 1 else SingleHPUStrategy(),
```
I would rather move `HPUParallelStrategy() if devices > 1 else SingleHPUStrategy()` out of the argument to improve redability

Reply:
+1